<?xml version="1.0" encoding="UTF-8"?>
<launch>

    
   <!--If you to teleoperate decomment this
  <param name="linear_scale" type="double" value="0.9" />
  <param name="angular_scale" type="double" value="0.9" />
  <param name="topic" type="str" value="cmd_vel" />-->

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="world_name" value="$(find projet)/worlds/Projet2023.world"/>
      <arg name="paused" value="false"/>
      <arg name="use_sim_time" value="true"/>
      <arg name="gui" value="true"/>
      <arg name="headless" value="false"/>
      <arg name="debug" value="false"/>
    </include>


  <!-- Spwan turtlebot in Gazebo -->
  <param name="robot_description" command="$(find xacro)/xacro $(find projet)/urdf/turtlebot3.urdf.xacro " />
  
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger -x 0.8 -y -1.74 -z 0.0 -Y 0 -param robot_description" /> 
  
  <!-- Launch the red line detector-->
  <node pkg="projet" name="Red_line_detector" type="Red_line_detector.py"/>
  <!-- Launch the robot controller, to run the right challenge-->                         
  <node pkg="projet" name="Robot_Brain" type="Robot_controller.py"/>

 <!--If you to teleoperate decomment this
  
  <node pkg="mybot_teleop" name="teleop" type="mybot_teleop.py" 
        output="screen" required="true">
      
  </node>	-->
  
</launch>
